### PR TITLE
Feat: Update user personal background

### LINKED ISSUE #49: Update User Personal Background
## Description
As a user,
I need to update the user personal background ,
so that I can change my personal background.

## Mocks
NA

### Update [Required]
- [ ] Create an "Update" functionality for user personal background page

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 2 mentors.

## Estimation
1 day

### PR DESCRIPTION
**Pull request by mtreacy002**
### Description
Allow user to update their personal background

Fixes #49 

### Type of Change:
- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- login then go to `My Space` > 'Personal Details' > `Personal Background`
- If this is the first time user view the Personal Background page, it will show the message below:
Screen Shot 2020-08-02 at 10 08 40 pm

- Fill in the personal background information. On successful process, see message below
Screen Shot 2020-08-04 at 8 34 52 am

If user click `Save` button without selecting anything, the default value is saved "Prefeer not to say"

Screen Shot 2020-08-04 at 8 35 19 am

- Now change the details and on successful update see message below

Screen Shot 2020-08-04 at 8 35 45 am

### Checklist:
- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by foongminwong — COMMENTED** [review]
Hi @mtreacy002 , I was able to login and go to the `Personal Background` page. 

test-pr-61-server-unavailable-when-login

_However_, it shows an error that `The server is currently unavailable. Try again later` after successful login while running both MS & BIT locally.

Prior to this, I did populate the personal background for user `bopogow` on Swagger UI (see below). So when I click on the `Personal Background` page, I should expect the page should have the populated information. Hmm any fix on this?

test-pr-61-already-populate-personal-bg

Other: Remember to resolve the conflicting files below

**Comment by mtreacy002** [comment]
Hi @foongminwong. I have resolved the merge conflict issue. Can you please try again by pulling the latest code change? Maybe that will fix the issue? Also, you don't need to create the personal background on backend bcoz the frontend should be able to create or update using the same `Save` button. 
Please let me know how you go.

**Comment by foongminwong** [comment]
> Hi @foongminwong. I have resolved the merge conflict issue. Can you please try again by pulling the latest code change? Maybe that will fix the issue? Also, you don't need to create the personal background on backend bcoz the frontend should be able to create or update using the same `Save` button.
> Please let me know how you go.

@mtreacy002  Hmm I pulled the latest codes but I still have the error `The server is currently unavailable. Try again later` for a user who has created their personal background (bg) on the backend.

I created the personal bg for user `bopogow` on the backend a while ago when tested [this PR](https://github.com/anitab-org/bridge-in-tech-backend/pull/83), so as a user, I assume their personal bg automatically GET from backend, when I clicked `Personal Background` on the frontend. Let me know your thoughts!

File: src/members/Members.jsx [hunk 1]
```diff
@@ -1,21 +1,79 @@
...
+    fetch(`${BASE_API}/users`, requestMembersList)
+      .then(async response => {
+        const data = await response.json();
+        if (response.ok) {
+          return setMembers(data);
+        }
+        setErrorMessage(data.message);
+      })
+      .catch(error =>
+        setErrorMessage("The service is temporarily unavailable, please try again later.")
```
Comment (meenakshi-dhanani) [inline src/members/Members.jsx]: Why does this not use the SERVICE_UNAVAILABLE_ERROR?

**Comment by meenakshi-dhanani** [comment]
Inside the Personal Background file, if you have already checked a is_public and the other value, then for any of the other fields, if !value then PreferNotToAnswer. 
you don't need to do that if with the other too many fields.

File: src/members/Members.jsx (hunk 1 above)
Comment (mtreacy002) [reply src/members/Members.jsx]: Yup. Fixed it.

**Comment by mtreacy002** [comment]
> > Hi @foongminwong. I have resolved the merge conflict issue. Can you please try again by pulling the latest code change? Maybe that will fix the issue? Also, you don't need to create the personal background on backend bcoz the frontend should be able to create or update using the same `Save` button.
> > Please let me know how you go.
> 
> @mtreacy002 Hmm I pulled the latest codes but I still have the error `The server is currently unavailable. Try again later` for a user who has created their personal background (bg) on the backend.
> 
> I created the personal bg for user `bopogow` on the backend a while ago when tested [this PR](https://github.com/anitab-org/bridge-in-tech-backend/pull/83), so as a user, I assume their personal bg automatically GET from backend, when I clicked `Personal Background` on the frontend. Let me know your thoughts!

@foongminwong, that should be the case (if data already there). I'm not sure why you have this issue since I wasn't able to replicate the bug at my end. Do you have time sometime today to meet on hangout to troubleshoot this with me? Leet me know when you're available.

**Comment by mtreacy002** [comment]
> Inside the Personal Background file, if you have already checked a is_public and the other value, then for any of the other fields, if !value then PreferNotToAnswer.
> you don't need to do that if with the other too many fields.

@meenakshi-dhanani . The reason why I need to specify each enum  fields since there are only 10 out 19 fields has default value "Prefer not to say". The other 8 default value is empty string. The is_public is the only boolean value.
Screen Shot 2020-08-04 at 8 35 19 am

**Comment by meenakshi-dhanani** [comment]
> > Inside the Personal Background file, if you have already checked a is_public and the other value, then for any of the other fields, if !value then PreferNotToAnswer.
> > you don't need to do that if with the other too many fields.
> 
> @meenakshi-dhanani . The reason why I need to specify each enum fields since there are only 10 out 19 fields has default value "Prefer not to say". The other 8 default value is empty string. The is_public is the only boolean value.
> Screen Shot 2020-08-04 at 8 35 19 am

What is the case for the other fields, in case their value is not present? I'm trying to see if there's a different answer for other fields that we use if their value is not available. Or can we generalize saying, if value is not there, then it is "Prefer not to answer". In that case you won't have to mention all fields

**Comment by mtreacy002** [comment]
@meenakshi-dhanani , I've checked and merged the PR you raised for the update personal background. It looks good and make sense. Thanks for your help. 😉

**Review by foongminwong — APPROVED** [review]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done

2. All possible responses were tested as below:

   * _Test PR 61 - Update user personal background_  
       _Screenshot/gif/url_:

   
    test-bit-web-pr-61-update-personal-bg-successfully

  
      _Expected Result_:  As a user, I should be able to update personal background.
      _Actual Result_: Same as expected.

3. OS Version: Windows 10

**Review by meenakshi-dhanani — APPROVED** [review]

**Pull request merged**